helper/schema/data_source_resource_shim.go

```diff
@@ -32,7 +32,7 @@ func DataSourceResourceShim(name string, dataSource *Resource) *Resource {
 
 	// FIXME: Link to some further docs either on the website or in the
 	// changelog, once such a thing exists.
-	dataSource.deprecationMessage = fmt.Sprintf(
+	dataSource.Deprecated = fmt.Sprintf(
```

Comment:
I think some of this feedback is expressed in Slack, but I think `DeprecationMessage` is the preferred var name. Other than that I see no reason to block this, we can build off it or keep it as our "Resource Deprecation" mechanism!

Reply:
Fixed!